Auth: Add skip_org_role_sync for AzureAD OAuth (#60322)
grafana/grafana: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md

```
force_use_graph_api = true
```

### Map roles

> By default, Azure AD authentication will map users to organization roles based on the most privileged application role assigned to the user in AzureAD.

Reviewer: ```suggestion
By default, Azure AD authentication maps users to organization roles based on the most privileged application role assigned to the user in AzureAD.
```

If no application role is found, the user is assigned the role specified by
[the `auto_assign_org_role` option]({{< relref "../../../configure-grafana#auto_assign_org_role" >}}).
You can disable this default role assignment by setting `role_attribute_strict = true`.
It denies user access if no role or an invalid role is returned.

> **On every login** the user organization role will be reset to match AzureAD's application role and

Reviewer: ```suggestion
On every login, the user organization role is reset to match AzureAD's application role and
```

> their organization membership will be reset to the default organization.

Reviewer: ```suggestion
their organization membership is reset to the default organization.
```

## Skip organization role sync

> If Azure AD authentication is not intended to sync user roles and organization membership,

Reviewer: ```suggestion
If you do not want Azure AD authentication to sync user roles and organization membership,
```

> `oauth_skip_org_role_update_sync` should be enabled, this is not recommended to use in favor of setting provider specific `skip_org_role_sync` option.

Reviewer: ```suggestion
then enable `oauth_skip_org_role_update_sync`. This is not the recommended setting. Use the provider-specific `skip_org_role_sync` option, when possible.
```

> See [configure-grafana]({{< relref "../../../configure-grafana#oauth_skip_org_role_update_sync" >}}) for more details.

Reviewer: ```suggestion
Refer to [configure-grafana]({{< relref "../../../configure-grafana#oauth_skip_org_role_update_sync" >}}) for more information.
```
